### PR TITLE
[ENH] Implement `flake8` and `pandas-vet`

### LINKED ISSUE #638: [INF] Consider `pandas-vet` suggestions
# Brief Description

I've been looking into the `pandas` linter [`pandas-vet`](https://github.com/deppen8/pandas-vet) which is a plugin for `flake8`. I ran it on the `pyjanitor` codebase to see if it had any suggestions. Here are the results:

```
janitor\chemistry.py:156:9: PD002 'inplace = True' should be avoided; it has inconsistent behavior
janitor\chemistry.py:157:5: PD002 'inplace = True' should be avoided; it has inconsistent behavior
janitor\engineering.py:104:21: PD011 Use '.array' or '.to_array()' instead of '.values'; 'values' is ambiguous
janitor\finance.py:332:24: PD011 Use '.array' or '.to_array()' instead of '.values'; 'values' is ambiguous
janitor\functions.py:286:38: PD005 Use arithmetic operator instead of method
janitor\functions.py:331:14: PD005 Use arithmetic operator instead of method
janitor\functions.py:332:12: PD005 Use arithmetic operator instead of method
janitor\functions.py:341:18: PD005 Use arithmetic operator instead of method
janitor\functions.py:389:24: PD003 '.isna' is preferred to '.isnull'; functionality is equivalent
janitor\functions.py:392:26: PD003 '.isna' is preferred to '.isnull'; functionality is equivalent
janitor\functions.py:2521:23: PD011 Use '.array' or '.to_array()' instead of '.values'; 'values' is ambiguous
janitor\functions.py:2526:20: PD011 Use '.array' or '.to_array()' instead of '.values'; 'values' is ambiguous
janitor\functions.py:2586:5: PD002 'inplace = True' should be avoided; it has inconsistent behavior
janitor\functions.py:2942:46: PD011 Use '.array' or '.to_array()' instead of '.values'; 'values' is ambiguous
janitor\functions.py:2992:15: PD003 '.isna' is preferred to '.isnull'; functionality is equivalent
janitor\functions.py:3556:48: PD003 '.isna' is preferred to '.isnull'; functionality is equivalent
janitor\io.py:48:22: PD011 Use '.array' or '.to_array()' instead of '.values'; 'values' is ambiguous
janitor\io.py:51:19: PD011 Use '.array' or '.to_array()' instead of '.values'; 'values' is ambiguous
janitor\io.py:57:31: PD011 Use '.array' or '.to_array()' instead of '.values'; 'values' is ambiguous
janitor\spark\functions.py:83:13: PD005 Use arithmetic operator instead of method
```

I haven't looked through all the warning to see if they are valid. Many seem to be about clearing up ambiguity. I'm not sure if this is worth adding into our pipeline officially, or just consider addressing some of these lines now.

Here's a list of the current warnings the plugin provides:
```
PD001: pandas should always be imported as 'import pandas as pd'

PD002: 'inplace = True' should be avoided; it has inconsistent behavior

PD003: '.isna' is preferred to '.isnull'; functionality is equivalent

PD004: '.notna' is preferred to '.notnull'; functionality is equivalent

PD005: Use arithmetic operator instead of method

PD006: Use comparison operator instead of method

PD007: '.ix' is deprecated; use more explicit '.loc' or '.iloc'

PD008: Use '.loc' instead of '.at'. If speed is important, use numpy.

PD009: Use '.iloc' instead of '.iat'. If speed is important, use numpy.

PD010 '.pivot_table' is preferred to '.pivot' or '.unstack'; provides same functionality

PD011 Use '.array' or '.to_array()' instead of '.values'; 'values' is ambiguous

PDO12 '.read_csv' is preferred to '.read_table'; provides same functionality

PD013 '.melt' is preferred to '.stack'; provides same functionality

PD015 Use '.merge' method instead of 'pd.merge' function. They have equivalent functionality.
```

### PR DESCRIPTION
**Pull request by hectormz**
# PR Description

Please describe the changes proposed in the pull request:

- This PR introduces `pandas-vet` for `pandas` linting, which is a `flake8` plugin
- `pycodestyle` is replaced with `flake8`, which includes `pycodestyle`, `pyflakes`, `mccabe`
    - Also updates pipeline and docuentation (PR template)
- Resolved _most_ `flake8` warnings
- Resolved _most_ `pandas-vet` warnings 

**This PR resolves #638 .**

## Remaining warnings:
```
.\janitor\finance.py:332:24: PD011 Use '.array' or '.to_array()' instead of '.values'; 'values' is ambiguous
.\janitor\functions.py:299:38: PD005 Use arithmetic operator instead of method
.\janitor\functions.py:344:14: PD005 Use arithmetic operator instead of method
.\janitor\functions.py:345:12: PD005 Use arithmetic operator instead of method
.\janitor\functions.py:354:18: PD005 Use arithmetic operator instead of method
.\janitor\functions.py:2511:23: PD011 Use '.array' or '.to_array()' instead of '.values'; 'values' is ambiguous
.\janitor\functions.py:2516:20: PD011 Use '.array' or '.to_array()' instead of '.values'; 'values' is ambiguous
.\janitor\functions.py:2576:5: PD002 'inplace = True' should be avoided; it has inconsistent behavior
.\janitor\io.py:48:22: PD011 Use '.array' or '.to_array()' instead of '.values'; 'values' is ambiguous
.\janitor\io.py:51:19: PD011 Use '.array' or '.to_array()' instead of '.values'; 'values' is ambiguous
.\janitor\io.py:58:18: PD011 Use '.array' or '.to_array()' instead of '.values'; 'values' is ambiguous
.\janitor\spark\functions.py:83:13: PD005 Use arithmetic operator instead of method
.\tests\finance\test_inflate_currency.py:55:5: F841 local variable 'df' is assigned to but never used
.\tests\finance\test_inflate_currency.py:73:5: F841 local variable 'df' is assigned to but never used
.\tests\functions\test_collapse_levels.py:26:9: PD011 Use '.array' or '.to_array()' instead of '.values'; 'values' is ambiguous
.\tests\functions\test_collapse_levels.py:32:9: PD011 Use '.array' or '.to_array()' instead of '.values'; 'values' is ambiguous
.\tests\functions\test_collapse_levels.py:48:9: PD011 Use '.array' or '.to_array()' instead of '.values'; 'values' is ambiguous
.\tests\functions\test_collapse_levels.py:58:9: PD011 Use '.array' or '.to_array()' instead of '.values'; 'values' is ambiguous
.\tests\functions\test_deconcatenate_column.py:47:9: F841 local variable 'df' is assigned to but never used
.\tests\functions\test_remove_columns.py:47:1: F811 redefinition of unused 'test_remove_columns_strange_cols' from line 36
.\tests\io\test_read_csvs.py:94:19: PD011 Use '.array' or '.to_array()' instead of '.values'; 'values' is ambiguous
.\tests\io\test_read_csvs.py:141:19: PD011 Use '.array' or '.to_array()' instead of '.values'; 'values' is ambiguous
```

For the most part, the remaining warnings are due to `pandas-vet` improperly assuming another variable type is a dataframe.

# PR Checklist

Please ensure that you have done the following:

1. [X] PR in from a fork off your branch. Do not PR from `<your_username>`:`dev`, but rather from `<your_username>`:`<feature-branch_name>`.

2. [X] If you're not on the contributors list, add yourself to `AUTHORS.rst`.

3. [X] Add a line to `CHANGELOG.rst` under the latest version header (i.e. the one that is "on deck") describing the contribution.
    - Do use some discretion here; if there are multiple PRs that are related, keep them in a single line.

## Quick Check

To do a very quick check that everything is correct, follow these steps below:

- [X] Run the command `make check` from pyjanitor's top-level directory. This will automatically run:
    - black formatting
    - flake8 checking
    - running the test suite
    - docs build

Once done, please check off the check-box above.

If `make check` does not work for you, you can execute the commands listed in the Makefile individually.

## Code Changes

If you are adding code changes, please ensure the following:

- [ ] Ensure that you have added tests.
- [X] Run all tests (`$ pytest .`) locally on your machine.
    - [X] Check to ensure that test coverage covers the lines of code that you have added.
    - [X] Ensure that all tests pass.

# Relevant Reviewers

Please tag maintainers to review.

- @ericmjl

### CODE REVIEW

**Comment by hectormz** [comment]
It's failing due to all the ongoing warnings mentioned above.

Some of the warnings I've addressed in recent issues.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -1,5 +1,6 @@
 new version (on deck)
 =====================
+- [INF]: Replace ``pycodestyle`` with ``flake8`` in order to add ``pandas-vet`` linter @hectormz
```
Comment (ericmjl) [inline CHANGELOG.rst]: :tada:

File: janitor/io.py [hunk 2]
```diff
@@ -32,28 +32,30 @@ def read_csvs(
     # Read the csv files
     # String to file/folder or file pattern provided
     if isinstance(files_path, str):
-        dfs = {
+        dfs_dict = {
```
Comment (ericmjl) [inline janitor/io.py]: Nice, being a bit more descriptive here.

**Comment by hectormz** [comment]
@ericmjl I can add in `#noqa` tags for the rest of the warnings that are due to `pandas-vet` misreading our variable types. I didn't add them in the event that we thought `pandas-vet` was not mature enough for us. I'll add them and push new commits

**Comment by hectormz** [comment]
Added new commits. This will also close #642 #639 and is a temporary bandage to #640

**Comment by hectormz** [comment]
@ericmjl all checks pass! Have a look over the recent changes and let me know if you have any comments or requests

**Comment by ericmjl** [comment]
Wonderful work as always, @hectormz!

No further comments here, I'm going to hit the big green button :smile:.

**Pull request merged**